connector/client/client.go

```diff
@@ -81,19 +82,50 @@ func (client *ConnectorClient) Post(ctx context.Context, target url.URL, activit
 	if err != nil {
 		return err
 	}
-	return client.sendRequest(req, activity)
+	return client.sendRequestWithRespErrCheck(req)
```

Comment:
I don't see any harm in having error check implicit in `sendRequest()` as we are always doing that

Reply:
What do you mean? I've separated `sendRequestWithRespErrCheck` from `sendRequest` as in case of `GET` request we're interested in the response. For other methods it make sense to return just the mapped error if that occurs.

Hope that make sense 👍 Please let me know if I misunderstood the comment.

Reply:
Okay, I think it's a good idea to separate them out.